IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,683 @@
+                                              context:(__unused id<MSIDRequestContext>)context
+                                                error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+// Remove one or more credentials from the keychain that match the key (see credentialItem:matchesKey).
+- (BOOL)removeItemsWithTokenKey:(__unused MSIDCacheKey *)key
+                        context:(__unused id<MSIDRequestContext>)context
+                          error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return NO;
+}
+
+#pragma mark - App Metadata
+
+// Save MSIDAppMetadataCacheItem (clientId/environment/familyId) in the macOS keychain cache.
+- (BOOL)saveAppMetadata:(__unused MSIDAppMetadataCacheItem *)item
+                    key:(__unused MSIDCacheKey *)key
+             serializer:(__unused id<MSIDAppMetadataItemSerializer>)serializer
+                context:(__unused id<MSIDRequestContext>)context
+                  error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return NO;
+}
+
+// Read MSIDAppMetadataCacheItem (clientId/environment/familyId) items from the macOS keychain cache.
+- (NSArray<MSIDAppMetadataCacheItem *> *)appMetadataEntriesWithKey:(__unused MSIDCacheKey *)key
+                                                        serializer:(__unused id<MSIDAppMetadataItemSerializer>)serializer
+                                                           context:(__unused id<MSIDRequestContext>)context
+                                                             error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+// Remove items with the given Metadata key from the macOS keychain cache.
+- (BOOL)removeItemsWithMetadataKey:(__unused MSIDCacheKey *)key
+                           context:(__unused id<MSIDRequestContext>)context
+                             error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return NO;
+}
+
+#pragma mark - Wipe Info
+
+// Saves information about the app which most-recently removed a token.
+- (BOOL)saveWipeInfoWithContext:(__unused id<MSIDRequestContext>)context
+                          error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return NO;
+}
+
+// Read information about the app which most-recently removed a token.
+- (NSDictionary *)wipeInfo:(__unused id<MSIDRequestContext>)context
+                     error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+#pragma mark - clear
+
+// A test-only method that deletes all items from the cache for the given context.
+- (BOOL)clearWithContext:(id<MSIDRequestContext>)context
```

Comment:
this currently only clears accounts. is the intention to also clear credentials?

Reply:
Yes, we'll add more here when add credentials support.